client-runtime/src/main/java/com/microsoft/rest/v2/http/NettyClient.java

```diff
@@ -91,11 +129,41 @@ public void channelCreated(Channel ch) throws Exception {
             }, eventLoopGroup.executorCount() * 2);
         }
 
+        @SuppressWarnings("unchecked")
         private NettyAdapter(int eventLoopGroupSize, int channelPoolSize) {
-            this.eventLoopGroup = new NioEventLoopGroup(eventLoopGroupSize);
+            MultithreadEventLoopGroup eventLoopGroup;
+            Class<? extends SocketChannel> channelClass;
+            try {
+                final String osName = System.getProperty("os.name");
+                if (osName.contains("Linux")) {
+                    eventLoopGroup = (MultithreadEventLoopGroup) Class.forName("io.netty.channel.epoll.EpollEventLoopGroup").getConstructor(Integer.TYPE).newInstance(eventLoopGroupSize);
```

Comment:
Make it more resilient towards maven hiccups? Maybe `NioEventLoopGroup` should be the fallback?

Reply:
Ahh I see it down there

Reply:
Writing a line to debug and using NioEventLoopGroup is the intended method to deal with a missing native transport. I guess the only reason to keep checking osName like this is to prevent spurious debug messages ("whoops! Couldn't find an epollEventLoopGroup! I don't care though, I'm on Windows...")

Reply:
You could change eventLoopGroupSize from `int` to `Integer`. That way if it's null, then you can assume the defaults and only look for the default constructor, not the constructor that takes an `int`.
Also, you should probably add `loadEventLoopGroup(String fullClassName)` and `loadSocketChanellClass(String fullClassName)` methods since your reflection stuff is identical except for the class name.